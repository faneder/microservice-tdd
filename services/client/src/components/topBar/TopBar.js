import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import {
  withStyles,
  AppBar,
  Toolbar,
  Typography,
  Button,
  IconButton,
} from '@material-ui/core';
import MenuIcon from '@material-ui/icons/Menu';

const styles = {
  root: {
    flexGrow: 1,
  },
  grow: {
    flexGrow: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
};

const items = [
  { key: '/', active: true, name: 'Home' },
  { key: '/about', name: 'About' },
  { key: '/user/status', name: 'User Status' },
];

const NavBar = (props) => {
  const { classes, title } = props;
  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <IconButton className={classes.menuButton} color="inherit" aria-label="Menu">
            <MenuIcon />
          </IconButton>
          <Typography variant="h6" color="inherit" className={classes.grow}>
            {title}
            { items.map((obj, key) =>
            <Button
              component={Link}
              name={obj.name}
              to={obj.key}
              key={key}
              color="inherit"
            >
              {obj.name}
            </Button>
            ) }
          </Typography>
          <Button color="inherit" component={Link} to='/login'>Login</Button>
          <Button color="inherit" component={Link} to='/logout'>Log Out</Button>
          <Button color="inherit" component={Link} to='/signup'>Sign Up</Button>
        </Toolbar>
      </AppBar>
    </div>
  );
}

NavBar.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(NavBar);
