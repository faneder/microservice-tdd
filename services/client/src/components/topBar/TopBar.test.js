import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import TopBar from './TopBar';

import {
  AppBar,
  Typography
} from '@material-ui/core';

const title = 'Title';

test('TopBar renders properly', () => {
  const wrapper = shallow(<TopBar title={title}/>);
  const element = wrapper.dive().find(AppBar)
  expect(element.length).toBe(1);
  expect(element.find(Typography).get(0).props.children[0]).toBe(title);
});

test('TopBar renders a snapshot properly', () => {
  const tree = renderer.create(
    <Router location="/">
      <TopBar title={title}/>
    </Router>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
