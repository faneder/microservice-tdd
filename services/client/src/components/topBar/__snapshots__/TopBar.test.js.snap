// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar renders a snapshot properly 1`] = `
<div
  className="NavBar-root-1"
>
  <header
    className="MuiPaper-root-13 MuiPaper-elevation4-19 MuiAppBar-root-4 MuiAppBar-positionStatic-8 MuiAppBar-colorPrimary-11"
  >
    <div
      className="MuiToolbar-root-40 MuiToolbar-regular-42 MuiToolbar-gutters-41"
    >
      <button
        aria-label="Menu"
        className="MuiButtonBase-root-50 MuiIconButton-root-44 MuiIconButton-colorInherit-45 NavBar-menuButton-3"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-49"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-53"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-124"
        />
      </button>
      <h6
        className="MuiTypography-root-62 MuiTypography-h6-79 MuiTypography-colorInherit-91 NavBar-grow-2"
      >
        Title
        <a
          className="MuiButtonBase-root-50 MuiButton-root-98 MuiButton-text-100 MuiButton-flat-103 MuiButton-colorInherit-119"
          href="/"
          name="Home"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-99"
          >
            Home
          </span>
          <span
            className="MuiTouchRipple-root-124"
          />
        </a>
        <a
          className="MuiButtonBase-root-50 MuiButton-root-98 MuiButton-text-100 MuiButton-flat-103 MuiButton-colorInherit-119"
          href="/about"
          name="About"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-99"
          >
            About
          </span>
          <span
            className="MuiTouchRipple-root-124"
          />
        </a>
        <a
          className="MuiButtonBase-root-50 MuiButton-root-98 MuiButton-text-100 MuiButton-flat-103 MuiButton-colorInherit-119"
          href="/user/status"
          name="User Status"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-99"
          >
            User Status
          </span>
          <span
            className="MuiTouchRipple-root-124"
          />
        </a>
      </h6>
      <a
        className="MuiButtonBase-root-50 MuiButton-root-98 MuiButton-text-100 MuiButton-flat-103 MuiButton-colorInherit-119"
        href="/login"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-99"
        >
          Login
        </span>
        <span
          className="MuiTouchRipple-root-124"
        />
      </a>
      <a
        className="MuiButtonBase-root-50 MuiButton-root-98 MuiButton-text-100 MuiButton-flat-103 MuiButton-colorInherit-119"
        href="/logout"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-99"
        >
          Log Out
        </span>
        <span
          className="MuiTouchRipple-root-124"
        />
      </a>
      <a
        className="MuiButtonBase-root-50 MuiButton-root-98 MuiButton-text-100 MuiButton-flat-103 MuiButton-colorInherit-119"
        href="/signup"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-99"
        >
          Sign Up
        </span>
        <span
          className="MuiTouchRipple-root-124"
        />
      </a>
    </div>
  </header>
</div>
`;
